src/main/java/org/springframework/data/redis/support/collections/DefaultRedisList.java

```diff
@@ -592,4 +593,15 @@ private void cap() {
 			listOps.trim(0, maxSize - 1);
 		}
 	}
+
+	@Override
+	public RedisList<E> reversed() {
+
+		DefaultRedisList<E> redisList = new DefaultRedisList<>(getKey(), getOperations(), this.maxSize);
+
+		StreamSupport.stream(this.spliterator(), false)
```

Comment:
We should honor that `reversed` is a view of the underlying list. Here, we duplicate the contents of the list by inserting everything that is stored at `getKey()`.

I suggest introducing a package-private `ReversedRedisList` type where we delegate to `this`. As our baseline remains Java 17, I think we will have to set up our own implementation without being able to reuse much from the Java libraries.

Looking at the JDK libs, their classes are delegations.

Reply:
Yeah, that was intentional and temporary hack. I will change this into a proper, view-based abstraction.